test/functional-api/zcommunications/notifications-application.it-spec.ts

```diff
@@ -0,0 +1,318 @@
+import {
+  assignUserToCommunity,
+  assignUserToCommunityVariablesData,
+} from '@test/utils/mutations/assign-mutation';
+import { removeChallenge } from '../integration/challenge/challenge.request.params';
+import { delay, getMailsData } from './communications-helper';
+
+let email = 'admin@alkem.io';
+let emailNonEco = 'non.ecoverse@alkem.io';
+let emailEcoAdmin = 'ecoverse.admin@alkem.io';
+let emailChallengeAdmin = 'ecoverse.member@alkem.io';
+let emailQAUser = 'qa.user@alkem.io';
+let globalAdmin = '';
+let nonEco = '';
+let ecoAdmin = '';
+let challengeAdmin = '';
+let qaUser = '';
+let ecoverseId = '';
+let organizationId = '';
+let ecoverseCommunityId = '';
+let communicationID = '';
+let challengeId = '';
+let challengeCommunityId = '';
+let applicationIdEcoverse = '';
+let ecoName = ecoverseName;
+let challengeName = `chName${uniqueId}`;
+
+beforeAll(async () => {
+  await deleteMailSlurperMails();
+
+  const responseOrg = await createOrganization(organizationName, hostNameId);
+  organizationId = responseOrg.body.data.createOrganization.id;
+
+  let responseEco = await createTestEcoverse(
+    ecoverseName,
+    ecoverseNameId,
+    organizationId
+  );
+  ecoverseId = responseEco.body.data.createEcoverse.id;
+  ecoverseCommunityId = responseEco.body.data.createEcoverse.community.id;
+  communicationID =
+    responseEco.body.data.createEcoverse.community.communication.id;
+
+  const responseChallenge = await mutation(
+    createChallenge,
+    challengeVariablesData(challengeName, `chnameid${uniqueId}`, ecoverseId)
+  );
+  challengeId = responseChallenge.body.data.createChallenge.id;
+  challengeCommunityId =
+    responseChallenge.body.data.createChallenge.community.id;
+
+  const requestUserData = await getUser(email);
+  globalAdmin = requestUserData.body.data.user.id;
+
+  const reqNonEco = await getUser(emailNonEco);
+  nonEco = reqNonEco.body.data.user.id;
+
+  const reqEcoAdmin = await getUser(emailEcoAdmin);
+  ecoAdmin = reqEcoAdmin.body.data.user.id;
+
+  const reqChallengeAdmin = await getUser(emailChallengeAdmin);
+  challengeAdmin = reqChallengeAdmin.body.data.user.id;
+
+  const reqQaUser = await getUser(emailQAUser);
+  qaUser = reqQaUser.body.data.user.id;
+
+  await mutation(
+    assignEcoverseAdmin,
+    userAsEcoverseAdminVariablesData(ecoAdmin, ecoverseId)
+  );
+
+  await mutation(
+    assignUserToCommunity,
+    assignUserToCommunityVariablesData(ecoverseCommunityId, challengeAdmin)
+  );
+
+  await mutation(
+    assignUserToCommunity,
+    assignUserToCommunityVariablesData(challengeCommunityId, challengeAdmin)
+  );
+
+  await mutation(
+    assignChallengeAdmin,
+    userAsChallengeAdminVariablesData(challengeAdmin, challengeId)
+  );
+});
+
+afterAll(async () => {
+  await removeChallenge(challengeId);
+  await removeEcoverse(ecoverseId);
+  await deleteOrganization(organizationId);
+});
+
+describe('Notifications - applications', () => {
+  beforeAll(async () => {
+    await mutation(
```

Comment:
Extract this into a method, like 'TriggerNotificationSettings' with input parameter boolean for true / false. You have copy / pasted the same code with only one difference, and it can be refactored in a method.